components/speedreader/rust/lib/src/readability/src/extractor.rs

```diff
@@ -140,12 +330,35 @@ pub fn post_process(dom: &mut Sink, root: Handle, meta: &Meta) {
             dom.append_before_sibling(&first_child, NodeOrText::AppendNode(title_header));
         }
         // Add in the description
-        if !meta.description.is_empty() {
-            let description =
-                dom::create_element_simple(dom, "p", "subhead metadata", Some(&meta.description));
+        if let Some(ref text) = meta.description {
+            let description = dom::create_element_simple(dom, "p", "subhead metadata", Some(text));
             dom.append_before_sibling(&first_child, NodeOrText::AppendNode(description));
         }
-        if !meta.title.is_empty() || !meta.description.is_empty() {
+
+        // Add in the author
+        if let Some(ref text) = meta.author {
+            let author =
+                dom::create_element_simple(dom, "p", "metadata", Some(&format!("By {}", text)));
+            dom.append_before_sibling(&first_child, NodeOrText::AppendNode(author));
+        }
+
+        // Add in last modified datetime
+        if meta.last_modified.is_some() {
+            let modified = dom::create_element_simple(
+                dom,
+                "p",
+                "metadata date",
+                Some(&meta.last_modified_to_string()),
+            );
+            dom.append_before_sibling(&first_child, NodeOrText::AppendNode(modified));
+        }
```

Comment:
Why not use the same `if let` syntax for `meta.last_modified` as you're using for `meta.author`? Then you won't even need that helper from above.

Reply:
fixed